lib/substitute.js

```diff
@@ -0,0 +1,36 @@
+
+/**
+ * Expose `substitute`
+ */
+
+module.exports = substitute;
+
+/**
+ * Type.
+ */
+
+var type = Object.prototype.toString;
+
+/**
+ * Substitute `:prop` with the given `obj` in `str`
```

Comment:
This isn't very clear IMO. The example in the project README on GitHub explains what this function does a lot more clearly, so if we're going to be removing the code from that context I think we should update the docs here to be a bit more descriptive. (Maybe with a usage example.)